network-api/networkapi/utility/middleware.py

```diff
@@ -12,6 +13,22 @@ class HttpResponseTemporaryRedirect(HttpResponseRedirectBase):
     status_code = 307
 
 
+class HTMLCleanUpMiddleware:
+    """
+    See http://blog.christianperone.com/2009/10/beautiful-django/
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
```

Comment:
decoding the response is side-stepping the gzip compression middleware. When the beautified HTML is set as the new value for `response.content` below, it's no longer gzip compressed, and clients choke trying to decode it. (Tested in Firefox and using httpie)

Reply:
We can either find a way to re-compress the payload (and recalculate the `Content-Length` value) or we can disable GZip compression when DEBUG is enabled.